File: crack.c
//~/pset2/crack
//crack.c
//pset2 cs50
//onelmv
//9.20.2018
//cracking password

#define _XOPEN_SOURCE
#include<unistd.h>
#include<stdio.h>
#include<string.h>
#include<cs50.h>




int main(int argc, string argv[])
{
    if (argc <= 2 && argc > 1) // Run if there are only two parameter.
    {
        char dic[52] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
                        'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
                        'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
                        'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R',
                        'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'
                       };

        string word  = NULL; // string for build each word
        string hash = argv[1]; //original hash
        char salt[2] = {argv[1][0], argv[1][1]};
        string myHash;  //my hash for compare

        for (int i = 0; i < 52; i++)
        {
            word = &dic[i];
            myHash = crypt(word, salt); //creating my hash

            if (strcmp(myHash, hash) == 0)
            {
                printf("%s\n", word);
                return 0;
            }
        }
        for (int i = 0; i < 52; i++)
        {
            for (int j = 0; j < 52; j++)
            {
                word[0] = dic[i];
                word[1] = dic[j];
                myHash = crypt(word, salt);


                if (strcmp(myHash, hash) == 0)
                {
                    printf("%s\n", word);
                    return 0;
                }

            }
        }
        for (int i = 0; i < 52; i++)
        {
            for (int j = 0; j < 52; j++)
            {
                for (int k = 0; k < 52; k++)
                {
                    word[0] = dic[i];
                    word[1] = dic[j];
                    word[2] = dic[k];
                    myHash = crypt(word, salt);
                    if (strcmp(myHash, hash) == 0)
                    {
                        printf("%s\n", word);
                        return 0;
                    }

                }
            }
        }
        for (int i = 0; i < 52; i++)
        {
            for (int j = 0; j < 52; j++)
            {
                for (int k = 0; k < 52; k++)
                {
                    for (int l = 0; l < 52; l++)
                    {
                        word[0] = dic[i];
                        word[1] = dic[j];
                        word[2] = dic[k];
                        word[3] = dic[l];
                        myHash = crypt(word, salt);
                        if (strcmp(myHash, hash) == 0)
                        {
                            printf("%s\n", word);
                            return 0;
                        }
                    }
                }
            }
        }

        for (int i = 0; i < 52; i++)
        {
            for (int j = 0; j < 52; j++)
            {
                for (int k = 0; k < 52; k++)
                {
                    for (int l = 0; l < 52; l++)
                    {
                        for (int m = 0; m < 52; m++)
                        {
                            word[0] = dic[i];
                            word[1] = dic[j];
                            word[2] = dic[k];
                            word[3] = dic[l];
                            word[4] = dic[m];
                            myHash = crypt(word, salt);
                            if (strcmp(myHash, hash) == 0)
                            {
                                printf("%s\n", word);
                                return 0;
                            }
                        }
                    }
                }
            }
        }
    }
    else
    {
        printf("Usage: ./crack hash\n"); // the arg have more than 1 parameter
        return 1;
    }
}//end

File: crack.py
# ~/pset6/crack
# crack.py
# pset6 cs50
# onelmv
# 12.28.2018
# cracking password

import sys
from crypt import crypt
from cs50 import get_string

if len(sys.argv) == 2:

    # Dictionary
    dic = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
           'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
           'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D',
           'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
           'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
           'Y', 'Z']

    # hash and salt
    h = sys.argv[1]
    salt = h[:2]

    # Cycle with 1 character word
    for i in dic:
        # checking if my hash is equal to the argument's hash
        if crypt(i, salt) == h:
            print(i)
            exit(0)

    # Cycle with 2 character word
    for i in dic:
        for j in dic:
            word = i + j
            if crypt(word, salt) == h:
                print(word)
                exit(0)

    # Cycle with 3 character word
    for i in dic:
        for j in dic:
            for k in dic:
                word = i + j + k
                if crypt(word, salt) == h:
                    print(word)
                    exit(0)

    # cycle with 4 character word
    for i in dic:
        for j in dic:
            for k in dic:
                for l in dic:
                    word = i + j + k + l
                    if crypt(word, salt) == h:
                        print(word)
                        exit(0)
else:
    print("Usage: ./crack hash")
    exit(1)